src/hooks/useBrowserPermissions.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
+
+export const useBrowserPermissions = (namePermission = 'geolocation') => {
+  const [permissionStatus, setPermissionStatus] = useState('denied');
+
+  useEffect(() => {
+    navigator?.permissions
+      ?.query({ name: namePermission })
```

Comment:
Looks like this isn't supported in Safari: https://developer.mozilla.org/en-US/docs/Web/API/Permissions/query

Reply:
@laurenmbeatty : Thank you! Yes, there isn't support for all browsers. My mistake. I'm sorry. I'm going to change to `navigator.location !== null` to identify if it's possible to share your location or not in your device, and my idea to know: if the user shared your location or not. Thanks!